### PR TITLE
Bukharney/feat(registration)/image inputbox

### PR DESCRIPTION
**Pull request by Bukharney**

### CODE REVIEW

**Comment by wit03** [comment]
LGTM Krub, But I think we should handle upload outside the component (the current version will handle it from inside of the component. @riflowth can I have your opinion?

**Comment by riflowth** [comment]
@wit03 I agree with u. But this code allows multiple files uploading but from the graphic requirement.

image

Seem we need to allow only a single file too.

File: apps/registration/src/components/ImageInputBox.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    const handleDragOver = (event:DragEvent)=>{
+        event.preventDefault()
+    }
+    const handleDrop = (event:any)=>{ 
+        event.preventDefault()
+        setFiles(event.dataTransfer.files)
+    }
+
+    if(files){
+        Array.from(files).map((file:any, idx) => {
```
Comment (wit03) [inline apps/registration/src/components/ImageInputBox.tsx:25]: Only one file is allowed to upload to this component.

File: apps/registration/src/components/ImageInputBox.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                        key = {idx}>{idx+1}<span>{". "}</span>{name}</li>
+                        }}   
+                    )}
+                </ul>
+            </div>
+            {!files && (
+                <div className={border}
+                onDragOver={handleDragOver}
+                onDrop={handleDrop}>
+                    <input type='file' multiple accept='.jpg, .pdf, .png' onChange={(event: any)=>{
```
Comment (wit03) [inline apps/registration/src/components/ImageInputBox.tsx:89]: onChange => onInput. see [Inputbox Component](https://github.com/cpe-kmutt-student/bangmod-hackathon-2023/blob/main/apps/registration/src/components/Inputbox.tsx) line 38 as a reference.

File: apps/registration/src/components/ImageInputBox.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import { useRef, useState } from 'preact/compat';
+
+export const ImageInputBox = (prop:{
+    name: string[]
+}) => {
+    const handleUpload = () => {}
+    const {name} = prop
+
+    //file in this obj
+    const [files, setFiles] = useState(null)
```
Comment (wit03) [inline apps/registration/src/components/ImageInputBox.tsx:10]: state and setState of state that will store file should be props which names obj and setObj. We will use provided props name "obj" to be the default value incase there is auto-save which will fetch the file value from API, and we will use props name "setObj" to set the value of obj.

**Review by wit03 — CHANGES_REQUESTED** [review]

File: apps/registration/src/components/ImageInputBox.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import { useRef } from 'preact/compat';
+
+export const ImageInputBox = ({
```
Comment (wit03) [inline apps/registration/src/components/ImageInputBox.tsx:3]: A little bit of change krub, please use `export default` so we don't need to import it as a module when we use it krub.

thanks

**Review by wit03 — CHANGES_REQUESTED** [review]

**Review by wit03 — APPROVED** [review]

**Pull request merged**